lib/utils.js

```diff
@@ -630,9 +630,23 @@ function _populateObj(obj) {
  */
 
 exports.getValue = function(path, obj, map) {
-  return mpath.get(path, obj, '_doc', map);
+  return mpath.get(path, obj, getValueLookup, map);
 };
 
+/*!
+ * ignore
+ */
+
+const mapGetterOptions = Object.freeze({ getters: false });
+
+function getValueLookup(obj, part) {
+  const _from = obj && obj._doc ? obj._doc : obj;
+  obj = _from instanceof Map ?
+    _from.get(part, mapGetterOptions) :
+    _from[part];
+  return obj;
```

Comment:
Is there a reason for reassigning it?

```suggestion
  return _from instanceof Map ?
    _from.get(part, mapGetterOptions) :
    _from[part];
```